### PR TITLE
CME-19 Spring Boot 3

### PR DESCRIPTION
**Pull request by rapolaskaseliscgi**
### Jira link
https://tools.hmcts.net/jira/browse/CME-19

### Change description
Spring Boot 3 upgrade

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/judicialapi/elinks/repository/ProfileRepository.java [hunk 1]
```diff
@@ -112,24 +112,18 @@ Page<UserProfile> fetchUserProfileByServiceNames(Set<String> ccdServiceCode,
...
+    @Query("select distinct per.objectId "
             + "from judicialUserProfile per "
             + "where (per.objectId != '' and per.objectId is not null)")
     List<String> fetchObjectId();
 
-    @Query(value = "select per "
-        + "from judicialUserProfile per "
-        + "where (per.objectId != '' and per.objectId is not null)"
-        + "and date(lastLoadedDate)=CURRENT_DATE ")
-    List<UserProfile> fetchObjectIdFromCurrentDate();
```
Comment (rapolaskaseliscgi) [inline src/main/java/uk/gov/hmcts/reform/judicialapi/elinks/repository/ProfileRepository.java:124]: This wasn't used anywhere, and it was failing hibernate validation.

File: src/functionalTest/java/uk/gov/hmcts/reform/judicialapi/idam/IdamOpenId.java [hunk 2]
```diff
@@ -0,0 +1,225 @@
+package uk.gov.hmcts.reform.judicialapi.idam;
```
Comment (rapolaskaseliscgi) [inline src/functionalTest/java/uk/gov/hmcts/reform/judicialapi/idam/IdamOpenId.java:1]: Copied over from `rd-commons-lib`

Comment (kiran-yenigala-hmcts) [reply src/functionalTest/java/uk/gov/hmcts/reform/judicialapi/idam/IdamOpenId.java:1]: why we need this class to be copied from rd-commons-lib, as we are already configured rd-commons-lib in build.gradle file

Comment (rapolaskaseliscgi) [reply src/functionalTest/java/uk/gov/hmcts/reform/judicialapi/idam/IdamOpenId.java:1]: Because that lib is not up to date and otherwise it would be blocking me. I removed that lib from build.gradle file.

Comment (kiran-yenigala-hmcts) [reply src/functionalTest/java/uk/gov/hmcts/reform/judicialapi/idam/IdamOpenId.java:1]: Ok..

**Review by kiran-yenigala-hmcts — DISMISSED** [review]

**Review by AntonyLeons — DISMISSED** [review]

File: src/functionalTest/java/uk/gov/hmcts/reform/judicialapi/AuthorizationFunctionalTest.java [hunk 3]
```diff
@@ -13,41 +14,21 @@
...
-    @Autowired
-    protected TestConfigProperties configProperties;
-
     protected static String s2sToken;
 
-    public static final String EMAIL = "EMAIL";
-    public static final String CREDS = "CREDS";
-    public static final String EMAIL_TEMPLATE = "test-user-%s@jrdfunctestuser.com";
-    public static final String ROLE_JRD_ADMIN = "jrd-admin";
     public static final String ROLE_JRD_SYSTEM_USER = "jrd-system-user";
```
Comment (SammiChong) [inline src/functionalTest/java/uk/gov/hmcts/reform/judicialapi/AuthorizationFunctionalTest.java:32]: This is unused can you remove it?

Comment (rapolaskaseliscgi) [reply src/functionalTest/java/uk/gov/hmcts/reform/judicialapi/AuthorizationFunctionalTest.java:32]: It actually is used in `JudicialUsersFunctionalTest` class...

**Review by kiran-yenigala-hmcts — APPROVED** [review]
LGTM

**Review by AntonyLeons — APPROVED** [review]

**Review by kiran-yenigala-hmcts — APPROVED** [review]
LGTM

File: src/main/java/uk/gov/hmcts/reform/judicialapi/configuration/SecurityConfiguration.java [hunk 4]
```diff
@@ -45,28 +46,18 @@ public class SecurityConfiguration {
...
+
     @Order(2)
     private final SecurityEndpointFilter securityEndpointFilter;
-    List<String> anonymousPaths;
-
-    private JwtAuthenticationConverter jwtAuthenticationConverter;
 
-    private RestAuthenticationEntryPoint restAuthenticationEntryPoint;
+    @Setter
+    @Getter
```
Comment (lukasz-wolski) [inline src/main/java/uk/gov/hmcts/reform/judicialapi/configuration/SecurityConfiguration.java:54]: anonymousPaths does not appear to be used outside SecurityConfiguration, do we need those Lombok generated accessor methods?

File: src/main/java/uk/gov/hmcts/reform/judicialapi/elinks/domain/BaseLocation.java [hunk 5]
```diff
@@ -24,43 +24,43 @@ public class BaseLocation {
...
     @Id
     @Column(name = "base_location_Id")
-    @JsonProperty(value = "id")
+    @JsonProperty("id")
     private String baseLocationId;
 
     @Column(name = "name")
     @Size(max = 256)
-    @JsonProperty(value = "name")
+    @JsonProperty("name")
```
Comment (lukasz-wolski) [inline src/main/java/uk/gov/hmcts/reform/judicialapi/elinks/domain/BaseLocation.java:32]: minor: name could be omitted, JsonProperty will default to name of field defined in class (same applies for all field annotations below)

File: src/main/java/uk/gov/hmcts/reform/judicialapi/service/impl/FeatureToggleServiceImpl.java [hunk 6]
```diff
@@ -30,8 +30,8 @@ public FeatureToggleServiceImpl(LDClient ldClient, @Value("${launchdarkly.sdk.us
         this.userName = userName;
     }
 
-    /**
-     * add controller.method name, flag name  in map to apply ld flag on api like below
+    /** mapServiceToFlag.
```
Comment (lukasz-wolski) [inline src/main/java/uk/gov/hmcts/reform/judicialapi/service/impl/FeatureToggleServiceImpl.java:33]: minor: please remove method name from javadoc, it will be assigned automatically

File: src/test/java/uk/gov/hmcts/reform/judicialapi/elinks/controller/ElinksControllerTest.java [hunk 7]
```diff
@@ -71,7 +71,7 @@ void test_load_location_success() {
 
         ResponseEntity<ElinkBaseLocationWrapperResponse> actual = eLinksController.loadLocation();
         assertThat(actual).isNotNull();
-        assertThat(actual.getStatusCodeValue()).isEqualTo(HttpStatus.OK.value());
+        assertThat(actual.getStatusCode().value()).isEqualTo(HttpStatus.OK.value());
```
Comment (lukasz-wolski) [inline src/test/java/uk/gov/hmcts/reform/judicialapi/elinks/controller/ElinksControllerTest.java:74]: minor: would be more readable to compare enum values to each other (same for other assertions below)

File: src/test/java/uk/gov/hmcts/reform/judicialapi/elinks/service/impl/ElinksPeopleServiceImplTest.java [hunk 8]
```diff
@@ -722,7 +721,7 @@ void loadPeopleForTribunalsWithParentIdEmptyWhiteSpaces() throws JsonProcessingE
                 .authorisationsRequests(List.of(AuthorisationsRequest.builder().jurisdiction("juristriction")
                         .ticket("lowerlevel").startDate("1991-12-19")
                         .endDate("2022-12-20").ticketCode("ticketId").build()))
-                .judiciaryRoles(Lists.newArrayList()).build();
+                .judiciaryRoles(List.of()).build();
```
Comment (lukasz-wolski) [inline src/test/java/uk/gov/hmcts/reform/judicialapi/elinks/service/impl/ElinksPeopleServiceImplTest.java:724]: minor: emptyList() imported statically from java.util.Collections would be more readable (same applies for other tests where an empty list is defined)

**Review by lukasz-wolski — APPROVED** [review]
please see comments inlined

File: build.gradle [hunk 9]
```diff
@@ -1,55 +1,64 @@
...
     }
 }
 
 plugins {
     id 'application'
+    id 'checkstyle'
+    id 'idea'
     id 'jacoco'
     id 'pmd'
+    id 'project-report'
```
[the remark this answers is not in the document]
Comment (rapolaskaseliscgi) [reply build.gradle:13]: To be fair, I've just copied from below. If you think it is not used, I am happy to remove.

File: src/main/java/uk/gov/hmcts/reform/judicialapi/service/impl/FeatureToggleServiceImpl.java (hunk 6 above)
Comment (rapolaskaseliscgi) [reply src/main/java/uk/gov/hmcts/reform/judicialapi/service/impl/FeatureToggleServiceImpl.java:33]: Removed method altogether, it is not needed.

**Review by lukasz-wolski — APPROVED** [review]

**Review by kiran-yenigala-hmcts — APPROVED** [review]

**Review by SammiChong — APPROVED** [review]

**Review by AntonyLeons — APPROVED** [review]
